test/typed_ecto_schema_test.exs

```diff
@@ -349,16 +363,36 @@ defmodule TypedEctoSchemaTest do
           normal: integer(),
           enforced: integer(),
           overriden: integer() | nil,
-          has_one: unquote(Ecto.Schema).has_one(HasOne.t()) | nil,
-          belongs_to: unquote(Ecto.Schema).belongs_to(BelongsTo.t()) | nil,
-          belongs_to_id: integer()
+          has_one: unquote(Ecto.Schema).has_one(HasOne.t()),
+          belongs_to: unquote(Ecto.Schema).belongs_to(BelongsTo.t()),
+          belongs_to_id: integer(),
+          has_many: unquote(Ecto.Schema).has_many(HasMany.t())
```

Comment:
Do we want to change the default behavior to be `null: false`?

Reply:
I'm confused, why without `null: false` the type doesn't have nulls but with that it suddenly have?